src/SFML/Graphics/Image.cpp

```diff
@@ -75,6 +76,27 @@ int eof(void* user)
     return stream.tell() >= stream.getSize();
 }
 
+// stb_image callbacks that operate on std::ifstream
+int readStdIfStream(void* user, char* data, int size)
+{
+    auto& stream = *static_cast<std::ifstream*>(user);
+    stream.read(data, size);
+    return static_cast<int>(stream.gcount());
+}
+
+void skipStdIfStream(void* user, int size)
+{
+    auto& stream = *static_cast<std::ifstream*>(user);
+    if (!stream.seekg(size, std::ios_base::cur))
+        sf::err() << "Failed to seek image loader std::ifstream" << std::endl;
+}
+
+int eofStdIfStream(void* user)
+{
+    auto& stream = *static_cast<std::ifstream*>(user);
+    return stream.eof();
+}
```

Comment:
Can you convert these to lambdas that are defined right before they're used?

Reply:
Yes, but old callbacks (for SFML stream class) do not do that. So do we make all callbacks be functions or convert the SFML ones to lambdas for consistency, or do we keep SFML callbacks as functions and add new ifstream ones as lambdas (so there's a mismatch)?

Reply:
Just make thew new ones lambdas. We can change the existing ones later if we want.

Reply:
I made all new load/save ones lambda.

I also added printing error reason in loadFromFile, from strerror, if opening file fails, so it will say something informative, not the "Image not of any known type, or corrupt" from STB.